roles/beats/tasks/main.yml

```
@@ -2,37 +2,38 @@
 
 - name: Prepare for whole stack roles if used
   when:
-  - elastic_stack_full_stack | bool
+    - elastic_stack_full_stack | bool
   block:
+    - name: Enable security if enabled for whole stack
+      set_fact:
+        beats_security: true
+      when:
+      - elastic_security | bool
+      - elastic_variant != "oss"
+      - not elastic_override_beats_tls | bool
 
-  - name: Enable security if enabled for whole stack
-    set_fact:
-      beats_security: true
-    when:
-    - elastic_security | bool
-    - elastic_variant != "oss"
-    - not elastic_override_beats_tls | bool
-
-  - name: Set elasticsearch_ca variable if not already done by user
-    set_fact:
-      elasticsearch_ca: "{{ groups['elasticsearch'][0] }}"
-    when:
-    - beats_security | bool
-    - elasticsearch_ca is undefined
-    tags:
-    - certificates
+    - name: Set elasticsearch_ca variable if not already done by user
+      set_fact:
+        elasticsearch_ca: "{{ groups['elasticsearch'][0] }}"
+      when:
+      - beats_security | bool
+      - elasticsearch_ca is undefined
 
-  - name: Set beats_ca_dir if whole stack is used
-    set_fact:
-      beats_ca_dir: "/etc/beats/certs"
-    when:
-    - elastic_security | bool
-    - beats_ca_dir is undefined
+    - name: Set beats_ca_dir if whole stack is used
+      set_fact:
+        beats_ca_dir: "/etc/beats/certs"
+      when:
+      - elastic_security | bool
+      - beats_ca_dir is undefined
 
-  - name: Set common password for common certificates
-    set_fact:
-      beats_tls_key_passphrase: "{{ elastic_cert_pass }}"
-    when: elastic_cert_pass is defined
+    - name: Set common password for common certificates
+      set_fact:
+        beats_tls_key_passphrase: "{{ elastic_cert_pass }}"
+      when: elastic_cert_pass is defined
```

```suggestion
      when: 
        - elastic_cert_pass is defined
        - beats_tls_key_passphrase is undefined
```
I guess syncing the two is only really neccessary when you use one single certificate for both services. As I understand, you cleaned that up. So I assume, it would work to have different passwords or even unique passwords per host.

---

Ahhh good notice, I updated it